pom.xml

```
@@ -177,5 +177,56 @@
       <url>https://repo.jenkins-ci.org/public/</url>
     </pluginRepository>
   </pluginRepositories>
-
+  <build>
+  <plugins>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-checkstyle-plugin</artifactId>
+      <version>3.1.1</version>
```

Please use the most recent release of the checkstyle plugin.
```suggestion
      <version>3.6.0</version>
```

---

Thank you for your previous comments and guidance,
I noticed that we don't currently have a checkstyle.xml file in the repository or a clear indication of where it might be referenced. I'm wondering if there is a plan to introduce this file, or if we're using an external/shared configuration that I might have overlooked? I ran mvn spotless:apply successfully, but just wanted to make sure we're not missing any custom style rules that might come with Checkstyle.

---

>  I'm wondering if there is a plan to introduce this file, or if we're using an external/shared configuration that I might have overlooked? I ran mvn spotless:apply successfully, but just wanted to make sure we're not missing any custom style rules that might come with Checkstyle.

I'm not planning to introduce checkstyle into this repository.  The Jenkins plugins that I maintain do not use checkstyle.  I've not found it to be valuable enough to be worth the effort to enable it.

If you're serious about enabling checkstyle in this repository, then you'll need to provide a complete implementation that includes the checkstyle settings to persuade me that it will be useful.  The default checkstyle settings conflict with spotless formatting.  I value spotless formatting much more than I value checkstyle warnings.  If your pull request proposes to accept many new, unresolved checkstyle warnings, then I'll reject it.  I'm not willing to add new warnings output because it will be a distraction and will create extra work for me to resolve the warnings.